applicationset/services/pull_request/bitbucket_cloud.go

```diff
@@ -0,0 +1,138 @@
+package pull_request
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/url"
+
+	bitbucket "github.com/ktrysmt/go-bitbucket"
+)
+
+type BitbucketCloudService struct {
+	client         	*bitbucket.Client
+	owner     	string
+	repositorySlug 	string
+}
+
+type BitbucketCloudPullRequest struct {
+	ID	int				`json:"id"`
+	Source 	BitbucketCloudPullRequestSource `json:"source"`
+}
+
+type BitbucketCloudPullRequestSource struct {
+	Branch BitbucketCloudPullRequestSourceBranch `json:"branch"`
+	Commit BitbucketCloudPullRequestSourceCommit `json:"commit"`
+}
+
+type BitbucketCloudPullRequestSourceBranch struct {
+	Name string `json:"name"`
+}
+
+type BitbucketCloudPullRequestSourceCommit struct {
+	Hash string `json:"hash"`
+}
+
+type PullRequestResponse struct {
+	Page		int32		`json:"page"`
+	Size 		int32 		`json:"size"`
+	Pagelen 	int32 		`json:"pagelen"`
+	Next 		string 		`json:"next"`
+	Previous 	string 		`json:"previous"`
+	Items 		[]PullRequest 	`json:"values"`
+}
+
+var _ PullRequestService = (*BitbucketCloudService)(nil)
+
+func parseUrl(uri string) (*url.URL, error) {
+	if uri == "" {
+		uri = "https://api.bitbucket.org/2.0"
+	}
+
+	url, err := url.Parse(uri)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing string %s to url: %v", uri, err)
```

Comment:
As we are enclosing the same error in the caller function, just returning `err` should be fine here.